python/pyiceberg/avro/reader.py

```diff
@@ -270,9 +270,8 @@ class DecimalReader(Reader):
     scale: int = dataclassfield()
 
     def read(self, decoder: BinaryDecoder) -> Decimal:
-        data = decoder.read(decoder.read_int())
-        unscaled_datum = int.from_bytes(data, byteorder="big", signed=True)
-        return unscaled_to_decimal(unscaled_datum, self.scale)
+        data = decoder.read(decimal_required_bytes(self.precision))
```

Comment:
`decimal_required_bytes(self.precision)` should be done in the constructor and reused, right? Or does a frozen dataclass make that hard?

Reply:
The frozen class makes it a bit awkward indeed, but did it anyway